### DISCUSSION TITLE
Not able to Deploy using .env

### DISCUSSION THREAD

**Discussion by abhinavreddyyy** [Cyfrin/foundry-full-course-cu#2586] [General]
When I try to use the .env file for private key and rpc url as $PRIVATE_KEY and $RPC_URL 
In terminal,
'''
abhinavreddy@Abhinav:~/foundry-test-project/hello_foundry/foundry-simple-storage-f23$ forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY
'''
 I'm getting the following error:
 error: a value is required for '--fork-url <URL>' but none was supplied
 
 I have checked my private key which has 64 character hexadecimal string with 0x and rpc url starting with http://

**Comment by AlimGlck** [comment]
What did u do exactly can u write one by one? bc I just ran into a problem about .env like u but my error message was different than u I solved that.

**Reply by abhinavreddyyy** [reply]
In the .env file first I wrote the private key as PRIVATE_KEY=0x..... and rpc url as RPC_URL=http://.. and saved it
in the vscode terminal, used source .env and checked using echo $PRIVATE_KEY 
then when i try to deploy using forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY 
I'm getting the same error as 
Failed to decode private key

**Reply by AlimGlck** [reply]
U must check the discussions first before posting a question in case the possibility that your problem is already solved.

Your problem is solved in this discussion 

https://github.com/Cyfrin/foundry-full-course-cu/discussions/2299

**Reply by AlimGlck** [reply]
But u will see in the next video we won't continue to use the .env file u can even skip that problem if u want.

**Comment by EngrPips** [comment]
Interesting, Great job here guys.

**Comment by Napcalx** [comment]
Run the source ,env file to initialize any new addition to the .env file before you run the command again
